### PR TITLE
Moarglob

### PR DESCRIPTION
**Pull request by wez**
* Fixup glob compatibility with python globbing (missing WM_PATHNAME) in one case
* Add a mode to de-dupe results for poorly written globs

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
By analyzing the blame information on this pull request, we identified @dhruvsinghal to be a potential reviewer.

File: query/eval.c [hunk 1]
```diff
@@ -468,8 +483,9 @@ static bool execute_common(struct w_query_ctx *ctx, w_perf_t *sample,
   if (w_perf_finish(sample)) {
     w_perf_add_root_meta(sample, ctx->lock->root);
     w_perf_add_meta(sample, "query_execute",
-                    json_pack("{s:b, s:i, s:i, s:O}",                   //
+                    json_pack("{s:b, s:i, s:i, s:i, s:O}",              //
                               "fresh_instance", res->is_fresh_instance, //
+                              "num_deduped", ctx->num_deduped,          //
```
Comment (bhamiltoncx) [inline query/eval.c]: I might be missing it, but I don't see anywhere that actually sets this value.

**Comment by bhamiltoncx** [comment]
Thanks! Looks like num_deduped isn't actually updated.

Did you add a test for deduping?

**Comment by facebook-github-bot-1** [comment]
@wez updated the pull request - [view changes](https://github.com/facebook/watchman/pull/352/files/d1ce5b9044af9621d2a004eea4a195ea779e1616..7c1fe0eeb61976b614f9012bdc7d3168cc2e7017)

File: query/eval.c (hunk 1 above)
Comment (wez) [inline query/eval.c]: heh, good catch :-p

**Comment by facebook-github-bot-0** [comment]
@wez updated the pull request - [view changes](https://github.com/facebook/watchman/pull/352/files/7c1fe0eeb61976b614f9012bdc7d3168cc2e7017..40691ff66afc48fd1f5a702b8bc04ac864d97b3f)

**Comment by wez** [comment]
Pushed as be21da2

**Pull request closed without merging**